chunk/dynamo_table_manager.go

```diff
@@ -0,0 +1,347 @@
+
+	var (
+		tablePeriodSecs = int64(m.cfg.TablePeriod / time.Second)
+		gracePeriodSecs = int64(m.cfg.CreationGracePeriod / time.Second)
+		maxChunkAgeSecs = int64(m.cfg.MaxChunkAge / time.Second)
+		firstTable      = m.cfg.PeriodicTableStartAt.Unix() / tablePeriodSecs
+		lastTable       = (mtime.Now().Unix() + gracePeriodSecs) / tablePeriodSecs
+		now             = mtime.Now().Unix()
+	)
+
+	// Add the legacy table
+	{
+		legacyTable := tableDescription{
+			name:             m.tableName,
+			provisionedRead:  m.cfg.ProvisionedReadThroughput,
+			provisionedWrite: minWriteCapacity,
+		}
+
+		// if we are before the switch to periodic table, we need to give this table write throughput
+		if now < (firstTable*tablePeriodSecs)+gracePeriodSecs+maxChunkAgeSecs {
+			legacyTable.provisionedWrite = m.cfg.ProvisionedWriteThroughput
+		}
+		result = append(result, legacyTable)
+	}
+
+	for i := firstTable; i <= lastTable; i++ {
+		table := tableDescription{
+			// Name construction needs to be consistent with chunk_store.bigBuckets
+			name:             m.cfg.TablePrefix + strconv.Itoa(int(i)),
+			provisionedRead:  m.cfg.ProvisionedReadThroughput,
+			provisionedWrite: minWriteCapacity,
+		}
+
+		// if now is within table [start - grace, end + grace), then we need some write throughput
+		if (i*tablePeriodSecs)-gracePeriodSecs <= now && now < (i*tablePeriodSecs)+tablePeriodSecs+gracePeriodSecs+maxChunkAgeSecs {
+			table.provisionedWrite = m.cfg.ProvisionedWriteThroughput
+		}
+		result = append(result, table)
+	}
+
+	sort.Sort(byName(result))
+	return result
+}
+
+func (m *DynamoTableManager) partitionTables(ctx context.Context, descriptions []tableDescription) ([]tableDescription, []tableDescription, error) {
+	// Get a list of tables that exist
+	existingTables := []string{}
+	if err := instrument.TimeRequestHistogram(ctx, "DynamoDB.ListTablesPages", dynamoRequestDuration, func(_ context.Context) error {
+		return m.dynamodb.ListTablesPages(&dynamodb.ListTablesInput{}, func(resp *dynamodb.ListTablesOutput, _ bool) bool {
+			for _, s := range resp.TableNames {
+				existingTables = append(existingTables, *s)
+			}
+			return true
+		})
+	}); err != nil {
+		return nil, nil, err
+	}
+	sort.Strings(existingTables)
+
+	// Work out tables that need to be created vs tables that need to be updated
+	toCreate, toCheckThroughput := []tableDescription{}, []tableDescription{}
+	i, j := 0, 0
+	for i < len(descriptions) && j < len(existingTables) {
+		if descriptions[i].name < existingTables[j] {
+			// Table descriptions[i] doesn't exist
+			toCreate = append(toCreate, descriptions[i])
+		} else if descriptions[i].name > existingTables[j] {
+			// existingTables[j].name isn't in descriptions, can ignore
+			j++
+		} else {
+			// Table existis, need to check it has correct throughput
+			toCheckThroughput = append(toCheckThroughput, descriptions[i])
+			i++
+			j++
+		}
+	}
+	for ; i < len(descriptions); i++ {
+		toCreate = append(toCreate, descriptions[i])
+	}
```

Comment:
Now we've broken out the listTables function, this is basically all thats left.  Still want me to do it?